src/devices/Dhtxx/DhtBase.cs

```diff
@@ -281,6 +281,10 @@ public void Dispose()
             {
                 _controller?.Dispose();
             }
+            else
+            {
+                _controller?.ClosePin(_pin);
+            }
 
             _i2cDevice?.Dispose();
```

Comment:
```suggestion
            _i2cDevice?.Dispose();
            _controller = null;
```
+ Need to make _controller non-readonly.

Reply:
if it's readonly, what is the advantage of making it non readonly and setting it up to null? Also if setup to null, it will have to go to line 287, not here.

Reply:
Because if it is not reset to null, the change to `_controller?.ClosePin(_pin)` doesn't change anything. We need to make sure that ClosePin is not called a second time, and this requires the _controller to be null the second time. Since we Dispose the controller in the first case of the `if`, we can set it to null in both cases. (Additional advantage: Using the class after dispose will throw exceptions).

Reply:
But in this case, we want to close the pin and **not** dispose the controller

Reply:
True, I didn't say it should be disposed in the second case. It should be set to null, but in both cases.